### PR TITLE
LPS-74324 More specific rule

### PR DESCRIPTION
**Pull request by rafaprax**

### CODE REVIEW

**Comment by rafaprax** [comment]
ci:retest

**Comment by rafaprax** [comment]
ci:retest

File: dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/css/_form-builder-field-types-modal.scss [hunk 1]
```diff
@@ -0,0 +1,69 @@
+.lfr-ddl-form-builder-field-types-modal {
```
Comment (leoadb) [inline dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/css/_form-builder-field-types-modal.scss]: @rafaprax Rename

File: dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/css/_form-builder-field-types-modal.scss [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	height: 100%;
+	margin: 0;
+	outline: none;
+	width: 100%;
+
+	&.lfr-ddl-form-builder-field-types-modal-hidden {
```
Comment (leoadb) [inline dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/css/_form-builder-field-types-modal.scss]: @rafaprax Rename

File: dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/css/_form-builder-field-types-modal.scss [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+				font-weight: 500;
+				line-height: initial;
+				margin: 8px 0 0 0;
+			}
+		}
+	}
+}
+
+@include sm {
+	.lfr-ddl-form-builder-field-types-modal {
```
Comment (leoadb) [inline dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/css/_form-builder-field-types-modal.scss]: @rafaprax Rename

File: dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/css/_form-builder-field-types-modal.scss [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+		max-height: 550px;
+		width: 392px;
+	}
+}
+
+@include md {
+	.lfr-ddl-form-builder-field-types-modal {
```
Comment (leoadb) [inline dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/css/_form-builder-field-types-modal.scss]: @rafaprax Rename

File: dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/css/_form-rules.scss [hunk 5]
```diff
@@ -0,0 +1,363 @@
+.liferay-ddl-form-rule-builder {
```
Comment (leoadb) [inline dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/css/_form-rules.scss]: @rafaprax Rename

File: dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/css/_form-rules.scss [hunk 6]
```diff
@@ -0,0 +1,363 @@
...
+	.ddl-form-basic-info {
```
Comment (leoadb) [inline dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/css/_form-rules.scss]: @rafaprax Rename

File: dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/css/_form-rules.scss [hunk 7]
```diff
@@ -0,0 +1,363 @@
...
+		display: none;
+	}
+
+	.ddl-form-builder-app {
```
Comment (leoadb) [inline dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/css/_form-rules.scss]: @rafaprax Rename

File: dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/css/_form-rules.scss [hunk 8]
```diff
@@ -0,0 +1,363 @@
...
+	}
+
+	.footer-button {
+		margin-left: 56px;
+		margin-top: 12px;
+		padding: 12px 24px;
+	}
+}
+
+.liferay-ddl-form-rule-builder-timeline {
```
Comment (leoadb) [inline dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/css/_form-rules.scss]: @rafaprax Rename

File: dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/css/_form-rules.scss [hunk 9]
```diff
@@ -0,0 +1,363 @@
...
+		position: absolute;
+		right: 0;
+		top: 0;
+
+		&:hover {
+			color: #65B6F0;
+		}
+	}
+
+	&.liferay-ddl-form-builder-rule-action-list {
```
Comment (leoadb) [inline dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/css/_form-rules.scss]: @rafaprax Rename

File: dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/css/_form-rules.scss [hunk 10]
```diff
@@ -0,0 +1,363 @@
...
+				padding-top: 45px;
+			}
+		}
+
+		.calculate-container-calculator-component {
+			min-width: 238px;
+			padding-bottom: 30px;
+			padding-top: 30px;
+
+			.liferay-ddl-form-builder-calculator {
```
Comment (leoadb) [inline dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/css/_form-rules.scss]: @rafaprax Rename

File: dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/css/_form-rules.scss [hunk 11]
```diff
@@ -0,0 +1,363 @@
...
+			right: 0;
+			top: 0;
+		}
+
+		button {
+			padding: 4px 12px;
+		}
+	}
+
+	&.liferay-ddl-form-builder-rule-condition-list {
```
Comment (leoadb) [inline dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/css/_form-rules.scss]: @rafaprax Rename

File: dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/css/_form-rules.scss [hunk 12]
```diff
@@ -0,0 +1,363 @@
...
+			}
+
+			.panel-inline {
+				display: inline-block;
+			}
+		}
+	}
+}
+
+.liferay-ddl-form-rule-builder-footer {
```
Comment (leoadb) [inline dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/css/_form-rules.scss]: @rafaprax Rename

File: dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/css/_form-rules.scss [hunk 13]
```diff
@@ -0,0 +1,363 @@
...
+	margin: $baseline * 3 0;
+	padding: 12px 54px;
+}
+
+.liferay-ddl-form-builder-rule-content {
```
Comment (leoadb) [inline dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/css/_form-rules.scss]: @rafaprax Rename

File: dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/css/_general.scss [hunk 14]
```diff
@@ -0,0 +1,345 @@
+.cursor-pointer {
+	cursor: pointer;
+}
+
+.ddl-form-builder-buttons .icon-spinner,
```
Comment (leoadb) [inline dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/css/_general.scss]: @rafaprax Rename

File: dynamic-data-mapping-form-builder/src/main/java/com/liferay/dynamic/data/mapping/form/builder/internal/converter/AutoFillDDLFormRuleActionFactory.java [hunk 15]
```diff
@@ -0,0 +1,110 @@
+/**
```
Comment (leoadb) [inline dynamic-data-mapping-form-builder/src/main/java/com/liferay/dynamic/data/mapping/form/builder/internal/converter/AutoFillDDLFormRuleActionFactory.java]: @rafaprax Rename

File: dynamic-data-mapping-form-taglib/liferay-form.xml [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+
+<taglibs osgi-module="true" short-name="liferay-form" tlib-version="1.0" uri="http://liferay.com/tld/form">
+	<authors>
+		<author>Rafael Praxedes</author>
+	</authors>
+	<component bodyContent="true" name="FormBuilder" writeJSP="true">
+		<attributes>
+			<attribute>
+				<name>ddmForm</name>
+				<inputType>com.liferay.dynamic.data.mapping.model.DDMForm</inputType>
```
Comment (leoadb) [inline dynamic-data-mapping-form-taglib/liferay-form.xml]: @rafaprax Use ddmStructureId

File: dynamic-data-mapping-form-builder/src/main/java/com/liferay/dynamic/data/mapping/form/builder/context/DDMFormContextDeserializer.java [hunk 17]
```diff
@@ -0,0 +1,31 @@
...
+import com.liferay.portal.kernel.exception.PortalException;
+
+/**
+ * @author Rafael Praxedes
+ */
+@ProviderType
+public interface DDMFormContextDeserializer<T> {
+
+	public T deserialize(
+			DDMFormContextDeserializerRequest contextDeserializerRequest)
```
Comment (leoadb) [inline dynamic-data-mapping-form-builder/src/main/java/com/liferay/dynamic/data/mapping/form/builder/context/DDMFormContextDeserializer.java]: @rafaprax Rename ddmFormContextDeserializerRequest

**Pull request merged**